delta/testkit/src/main/scala/ch/epfl/bluebrain/nexus/testkit/MonixBioAssertions.scala

```diff
@@ -21,26 +21,33 @@ trait MonixBioAssertions { self: Assertions =>
 
   implicit class IoAssertionsOps[E, A](io: IO[E, A])(implicit E: ClassTag[E], loc: Location) {
 
-    def assert(expected: A, clue: Any = "values are not the same"): UIO[Unit] = io.attempt.map {
-      case Left(NonFatal(err)) =>
-        val baos  = new ByteArrayOutputStream()
-        err.printStackTrace(new PrintStream(baos))
-        val stack = new String(baos.toByteArray)
-        fail(
-          s"""Error caught of type '${err.getClass.getName}', expected a successful response
-             |Message: ${err.toString}
-             |Stack:
-             |$stack""".stripMargin,
-          err
-        )
-      case Left(err)           =>
-        fail(
-          s"""Error caught of type '${E.runtimeClass.getName}', expected a successful response
-             |Message: ${err.toString}""".stripMargin
-        )
-      case Right(a)            => assertEquals(a, expected, clue)
+    private def exceptionHandler(err: Throwable) = {
+      val baos  = new ByteArrayOutputStream()
+      err.printStackTrace(new PrintStream(baos))
+      val stack = new String(baos.toByteArray)
+      fail(
+        s"""Error caught of type '${err.getClass.getName}', expected a successful response
+           |Message: ${err.getMessage}
+           |Stack:
+           |$stack""".stripMargin,
+        err
+      )
     }
 
+    def assert(expected: A, clue: Any = "values are not the same"): UIO[Unit] = io.redeemCause(
+      {
+        case Error(NonFatal(err)) =>
+          exceptionHandler(err)
```

Comment:
doesn't munit already print the cleaned stack trace when an error is thrown? is this necessary?

Reply:
I don't have it anymore and maybe it was because it was because of the `EvaluationError` that wasn't defined properly.
In any case, there is also an attempt here to have messages that are consistent between typed and terminal errors.